File: src/main/java/com/indra87g/commands/ReloadCommand.java
package com.indra87g.commands;

import cn.nukkit.Server;
import cn.nukkit.plugin.Plugin;
import cn.nukkit.Player;
import com.indra87g.utils.MessageHandler;

public class ReloadCommand extends BaseCommand {

    public ReloadCommand() {
        super("reload", "Reload a plugin", "/reload <pluginName>", "waffle.reload");
    }

    @Override
    protected boolean validateArgs(String[] args, Player player) {
        if (args.length != 1) {
            MessageHandler.sendMessage(player, "reload_usage");
            return false;
        }
        return true;
    }

    @Override
    protected boolean handleCommand(Player player, String[] args) {
        String pluginName = args[0];
        Plugin plugin = Server.getInstance().getPluginManager().getPlugin(pluginName);

        if (plugin == null) {
            MessageHandler.sendMessage(player, "reload_not_found", "{plugin}", pluginName);
            return true;
        }

        try {
            Server.getInstance().getPluginManager().disablePlugin(plugin);
            Server.getInstance().getPluginManager().enablePlugin(plugin);
            MessageHandler.sendMessage(player, "reload_success", "{plugin}", pluginName);
        } catch (Exception e) {
            MessageHandler.sendMessage(player, "reload_error", "{plugin}", pluginName);
            e.printStackTrace();
        }
        return true;
    }
}

File: src/main/kotlin/com/indra87g/commands/ReloadCommand.kt
package com.indra87g.commands

import cn.nukkit.Player
import cn.nukkit.Server
import com.indra87g.utils.MessageHandler

class ReloadCommand : BaseCommand("reload", "Reload a plugin", "/reload <pluginName>", "waffle.reload") {

    override fun validateArgs(args: Array<String>, player: Player): Boolean {
        if (args.size != 1) {
            MessageHandler.sendMessage(player, "reload_usage")
            return false
        }
        return true
    }

    override fun handleCommand(player: Player, args: Array<String>): Boolean {
        val pluginName = args[0]
        val plugin = Server.getInstance().pluginManager.getPlugin(pluginName)

        if (plugin == null) {
            MessageHandler.sendMessage(player, "reload_not_found", "{plugin}", pluginName)
            return true
        }

        try {
            Server.getInstance().pluginManager.disablePlugin(plugin)
            Server.getInstance().pluginManager.enablePlugin(plugin)
            MessageHandler.sendMessage(player, "reload_success", "{plugin}", pluginName)
        } catch (e: Exception) {
            MessageHandler.sendMessage(player, "reload_error", "{plugin}", pluginName)
            e.printStackTrace()
        }
        return true
    }
}
